File: app/src/main/java/peter/util/searcher/net/CommonRetrofit.java
package peter.util.searcher.net;


import com.facebook.stetho.okhttp3.StethoInterceptor;

import java.io.File;
import java.util.concurrent.TimeUnit;

import okhttp3.Cache;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import peter.util.searcher.BuildConfig;
import peter.util.searcher.Searcher;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class CommonRetrofit {

    private Retrofit retrofit;
    private static final String URL = "http://searcher-1254131086.file.myqcloud.com/";
    private static final String CACHE_NAME = "engines";
    private static final long MAX_CACHE = 1024 * 1024 * 10; //10M

    private CommonRetrofit() {
    }

    private static class SingletonInstance {
        private static final CommonRetrofit INSTANCE = new CommonRetrofit();
    }

    public static CommonRetrofit getInstance() {
        return SingletonInstance.INSTANCE;
    }

    public Retrofit getRetrofit() {
        if (retrofit == null) {
            OkHttpClient.Builder builder = new OkHttpClient.Builder();

            if (BuildConfig.DEBUG) {
                // Log信息拦截器
                HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
                loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);//这里可以选择拦截级别
                //设置 Debug Log 模式
                builder.addInterceptor(loggingInterceptor);
            }

            OkHttpClient okHttpClient = builder
                    .connectTimeout(30, TimeUnit.SECONDS)
                    .readTimeout(30, TimeUnit.SECONDS)
                    .writeTimeout(30, TimeUnit.SECONDS)
                    .cache(new Cache(new File(Searcher.context.getCacheDir(), CACHE_NAME), MAX_CACHE))
                    .build();

            if (BuildConfig.DEBUG) {
                builder.addNetworkInterceptor(new StethoInterceptor());
            }

            retrofit = new Retrofit.Builder()
                    .baseUrl(URL)
                    .client(okHttpClient)
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .build();
        }
        return retrofit;
    }


}

File: app/src/main/java/peter/util/searcher/net/CommonRetrofit.kt
package peter.util.searcher.net


//import com.facebook.stetho.okhttp3.StethoInterceptor;

import java.io.File
import java.util.concurrent.TimeUnit

import okhttp3.Cache
import okhttp3.OkHttpClient
import peter.util.searcher.Searcher
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class CommonRetrofit private constructor() {

    private var retrofit: Retrofit? = null

    private object SingletonInstance {
        val INSTANCE = CommonRetrofit()
    }

    companion object {
        private val URL = "http://7xoxmg.com1.z0.glb.clouddn.com/"
        private val CACHE_NAME = "engines"
        private val MAX_CACHE = (1024 * 1024 * 10).toLong() //10M
        val instance: CommonRetrofit by lazy { SingletonInstance.INSTANCE }
    }

    fun getRetrofit(): Retrofit? {
        if (retrofit == null) {
            val okHttpClient = OkHttpClient.Builder()
                    .connectTimeout(30, TimeUnit.SECONDS)
                    .readTimeout(30, TimeUnit.SECONDS)
                    .writeTimeout(30, TimeUnit.SECONDS)
                    .cache(Cache(File(Searcher.context!!.cacheDir, CACHE_NAME), MAX_CACHE))
                    //                    .addNetworkInterceptor(new StethoInterceptor())
                    .build()
            //            Gson gson = new GsonBuilder().setLenient().create();
            retrofit = Retrofit.Builder()
                    .baseUrl(URL)
                    .client(okHttpClient)
                    //                    .addConverterFactory(GsonConverterFactory.create(gson))
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .build()
        }
        return retrofit
    }


}